net/headers.go

```diff
@@ -42,12 +48,28 @@ func (h *ForwardedHeaders) Set(req *http.Request) {
 		req.Header.Set("X-Forwarded-Host", req.Host)
 	}
 
+	if h.Method {
+		req.Header.Set("X-Forwarded-Method", req.Method)
+	}
+
+	if h.Uri {
+		req.Header.Set("X-Forwarded-Uri", req.RequestURI)
+	}
+
 	if h.Port != "" {
 		req.Header.Set("X-Forwarded-Port", h.Port)
 	}
 
 	if h.Proto != "" {
 		req.Header.Set("X-Forwarded-Proto", h.Proto)
+
+		if h.OriginalURL {
```

Comment:
What would be the difference with `req.URL.String()`?

Reply:
h.OriginalURL is a bool or do you comment on the wrong line?

Reply:
I mean can we use `req.URL.String()` instead of `h.Proto+"://"+req.Host+":"+h.Port+req.RequestURI`?

Reply:
Also `req.Host` can contain `host:port` so the else branch may produce incorrect URL.

Reply:
The comment https://github.com/authelia/authelia/issues/3492#issuecomment-1150505829 suggests that 
> Probably best to ignore the X-Original-URL in that case

so maybe we just have `X-Forwarded-Uri` if it is enough and not bother reconstructing URL